internal/resource-request-operator/broadcaster.go

```diff
@@ -147,36 +149,86 @@ func (b *Broadcaster) onNodeDelete(obj interface{}) {
+			klog.V(4).Infof("OnPodAdd: Pod %s:%s passed ClusterID check. ClusterID = %s\n", podAdded.Namespace, podAdded.Name, clusterID)
+			currentPodsResources := b.readPodResources(clusterID)
+			// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+			// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+			addResources(currentPodsResources, podResources)
+			b.writePodsResources(clusterID, currentPodsResources)
+		}
 	}
 }
 
-func (b *Broadcaster) onPodDelete(obj interface{}) {
-	pod := obj.(*corev1.Pod)
-	klog.V(4).Infof("OnPodDelete: Delete for pod %s:%s\n", pod.Namespace, pod.Name)
-	podResources := extractPodResources(pod)
+func (b *Broadcaster) onPodUpdate(oldObj, newObj interface{}) {
+	oldPod := oldObj.(*corev1.Pod)
+	newPod := newObj.(*corev1.Pod)
+	klog.V(4).Infof("OnPodUpdate: Update for pod %s:%s\n", newPod.Namespace, newPod.Name)
+	newResources := extractPodResources(newPod)
+	oldResources := extractPodResources(oldPod)
 	currentResources := b.readClusterResources()
-	// this action is the complementary of the onPodAdd() one. For more details see at that function.
-	addResources(currentResources, podResources)
+	if pod.IsPodReady(newPod) {
+		// pod is already Ready, try to update resources.
+		if pod.IsPodReady(oldPod) {
+			updateResources(currentResources, oldResources, newResources)
+			if clusterID := newPod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+				klog.V(4).Infof("OnPodUpdate: Pod %s:%s passed ClusterID check. ClusterID = %s\n", newPod.Namespace, newPod.Name, clusterID)
+				currentPodsResources := b.readPodResources(clusterID)
+				// update the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+				// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+				updateResources(currentPodsResources, oldResources, newResources)
+				b.writePodsResources(clusterID, currentPodsResources)
+			}
+			// pod is becoming ready, same of onPodAdd case.
+		} else {
+			subResources(currentResources, newResources)
+			if clusterID := newPod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+				klog.V(4).Infof("OnPodUpdate: Pod %s:%s passed ClusterID check. ClusterID = %s\n", newPod.Namespace, newPod.Name, clusterID)
+				currentPodsResources := b.readPodResources(clusterID)
+				// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+				// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+				addResources(currentPodsResources, newResources)
+				b.writePodsResources(clusterID, currentPodsResources)
+			}
+		}
+		// pod is becoming not ready, same of onPodDelete case.
+	} else if pod.IsPodReady(oldPod) && !pod.IsPodReady(newPod) {
```

Comment:
This if statement is asymmetric: I would find it clearer having 3 similar conditions instead of nested ifs.